Productdetail

desafree opened the pull request:
Create the product detail ui. Content on page change based on the url

desafree committed:
rebase: rebase branch to main

desafree committed:
rebase: resolve conflict halfsection component

desafree committed:
refactor: implement review changes

-update props style usage
-remove redux dependency on components
-remove static image url
-add new text in i18n
-rename translate function

desafree committed:
feat: build product page image section

desafree committed:
fix: fix bug in halfsectionlist and grid component

-grid: pass only particular props if container or item.
-halfsectionlist: missing key

desafree committed:
feat: build product page product suggested section

desafree committed:
feat: add product page categories section

desafree committed:
feat: add to product page info section

desafree committed:
docs: remove .env file from git

desafree commented:
Ci sono un pò più di 20 file modificati perchè non volevo spezzare in due la PR

rams23 commented on src/_shared/components/Grid/Grid.tsx:
```diff
@@ -55,22 +55,32 @@ export const Grid: FC<Props> = ({
   alignItems = "stretch",
   className,
 }) => {
+  if (container) {
+    return (
+      <GridMUI
+        columnSpacing={columnSpacing}
+        container={container}
+        rowSpacing={rowSpacing}
+        spacing={spacing}
+        wrap={wrap}
+        direction={direction}
+        alignItems={alignItems}
+        justifyContent={justifyContent}
+        className={className}
+      >
+        {children}
+      </GridMUI>
+    );
+  }
+
   return (
     <GridMUI
```
come mai abbiamo dovuto dividere così?

desafree replied:
alcuni props (come columnSpacing o direction) sono specifici per Grid-container mentre altri (come sx o md) solo per Grid-item

rams23 commented on src/_shared/components/HalfSectionList/HalfSectionList.tsx:
```diff
@@ -1,9 +1,9 @@
+import { HalfSectionListStyled } from "./HalfSectionList.styled";
 
 interface Props {
   title1: string;
```
Megli usare nomi più espliciti e non title1 e title2

rams23 commented on src/_shared/components/HalfSectionList/HalfSectionList.tsx:
```diff
@@ -14,34 +14,37 @@ interface Props {
-                </Box>{" "}
-                <Typography variant="body1">{item.item}</Typography>
+              <ListItem
+                disableGutters={true}
+                disablePadding={true}
```
rimuovere i true

rams23 commented on src/products/components/ButtonProduct/ButtonProduct.styled.tsx:
```diff
@@ -0,0 +1,35 @@
+      color: rgba(0, 0, 0, 0.25);
+    }
+
+    p {
+      font-family: "Manrope";
+      padding: 20px;
+      color: black;
+      font-size: 13px;
+      font-weight: 700;
+      line-height: 18px;
+      letter-spacing: 1px;
+      text-align: center;
```
si può usare Typography

rams23 commented on src/products/components/ButtonProduct/ButtonProduct.tsx:
```diff
@@ -0,0 +1,28 @@
+
+  return (
+    <ButtonProductStyled>
+      <div>
+        <button onClick={handleCLickMinus}>-</button>
+        <p>{qty}</p>
+        <button onClick={handleClickPlus}>+</button>
+      </div>
```
Meglio usare i componenti che abbiamo

rams23 commented on src/products/components/ProductDetail/ProductDetail.tsx:
```diff
@@ -0,0 +1,32 @@
+
+  return (
+    <ProductDetailStyled>
+      <button onClick={handleClick}>Go back</button>
```
component e traslations

rams23 commented on src/products/components/ProductImages/ProductImages.styled.tsx:
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+
+interface Props {
+  imageList: string[];
```
se ci devono sempre essere treimmagini melgio usare [string, string, string], altrimenti potrei passare un array vuoto

desafree committed:
refactor: implement review changes

-add better names for props in halfsectionlist component
-remove ridondant {true} as props
-add new typography variant
-refactor buttonproduct component using pre-existing shared components
-fix type in imagelist

desafree committed:
refactor: replace text with typography component

desafree committed:
refactor: replace goback component with material ui link

desafree committed:
refactor: add translation go back in productdetail component
